CIRC-5023 Improve Web DB failover instructions (#58)
circonus/docs: content/circonus/on-premises/roles-services/web-db.md

In this directory there is also a `pg_log` directly, which will contain log files useful for general debugging of database connectivity issues. It also contains some queries.

Note that any config files located in the data directory are managed by
Circonus. Any manual changes will not persist if `run-hooper` is used. Some
common tuning parameters can be changed via
[site.json](/circonus/on-premises/installation/installation/#web-db-attributes).
If you need additional customizations beyond what Hooper provides, please
contact Circonus Support (support@circonus.com).

## Web DB Failover

In the event of a database failure, it will be necessary to manually failover
to one of your replicas, which will become the new primary.  To do this, use the following steps.

>  1. If the current primary is still running, shut it down.

Reviewer: ```suggestion
 1. If the current primary is still running, shut down the Postgres service.
```

Author: Will do that, but also give the service name.

>  1. On the new primary, make a copy of the file `/wdb/pgdata/9.2/recovery.conf`

Reviewer: Can we bold the "new" here, or somehow use better verbage throughout? Both are called the 'primary' and this tripped me up more than once while reading through it quickly.  I can only imagine someone in an emergency situation trying to go fast can mix these up easily.

Author: Can do.

    to a location outside of the `/wdb/pgdata` hierarchy, such as your home
    directory.
 1. If using an IP alias as the `connect_host`, relocate the IP/DNS name to the
    new primary. Otherwise, update `site.json`, setting a new hostname for
    `master` and `connect_host` in the `web_db` object.
 1. On the new primary, touch the file `/wdb/pgdata/9.2/failover.now`.
    * When the new primary completes its failover, the file `/wdb/pgdata/9.2/recovery.conf` will be renamed to "`recovery.done`".

>     * After the renaming occurs, you can delete the `failover.now` file.

Reviewer: I think the failover procedure removed this for us.  I don't remember doing this step.

Author: I've added "if it is present" at the end of that sentence.

 1. Run Hooper to ensure any configuration customizations are applied:

>     `/opt/circonus/bin/run-hooper -m`, restarting the database if necessary.

Reviewer: ```suggestion
    `/opt/circonus/bin/run-hooper -m`  This command will restart the database if necessary.
```

Author: Hooper will not auto-restart the database in the current 9.2 setup.

 1. Distribute the updated `site.json` to all other hosts in your deployment,

>     and perform a Hooper run across all hosts, following the order specified in

Reviewer: ```suggestion
    and perform a Hooper run (`/opt/circonus/bin/run-hooper -m`) across all hosts, following the order specified in
```

    the [initial installation](/circonus/on-premises/installation/installation/#installation-sequence)

>     page.

Reviewer: ```suggestion
    page.
  1.  At this point your installation should be failed over and serving traffic from the new primary.
```

>  1. Rebuild the other replicas. Follow these steps on each machine:

Reviewer: ```suggestion
 1. Rebuild any other replicas off of the new primary.  Follow these steps on each machine:
```

    1. Stop the postgres service.
    1. Run the command: `rm -rf /wdb/pgdata/9.2`
    1. Run the command: `mkdir /wdb/pgdata/9.2`
    1. Run the command: `chown postgres:postgres /wdb/pgdata/9.2`
    1. Run the command: `chmod 700 /wdb/pgdata/9.2`
    1. Run the command: `sudo -u postgres /opt/pgsql/bin/pg_basebackup -D /wdb/pgdata/9.2/ -h <new primary IP> -U replication`
    1. Edit the `recovery.conf` file from Step 2, changing the host in the primary_conninfo parameter to point to the new primary.
    1. Place the `recovery.conf` into the path `/wdb/pgdata/9.2/recovery.conf`, overwriting the current file in that path.
    1. Start the postgres service.

>  1. When the old primary comes back online, make it into a replica by running

Reviewer: ```suggestion
 1. When/if the problems with the old primary are addressed, make it into a replica by running
```

    the rebuild commands from the previous step on that machine.

> You should now have a new primary DB and services should reconnect to it when needed.  At this point, refer to the [Service Dependencies](/circonus/on-premises/service-dependencies) section for a list of services that we recommend restarting.

Reviewer: Should we move this segment to before the replicas to replace my suggestion above?

Author: Hooper will actually notify the operator of the specific services that need to be restarted, at the end of the run.  I think we can just delete this.
